src/node_errors.cc

```diff
@@ -493,6 +493,36 @@ void OnFatalError(const char* location, const char* message) {
   ABORT();
 }
 
+void OOMErrorHandler(const char* location, bool is_heap_oom) {
+  const char* message =
+      is_heap_oom ? "Allocation failed - JavaScript heap out of memory"
+                  : "Allocation failed - process out of memory";
+  if (location) {
+    FPrintF(stderr, "FATAL ERROR: %s %s\n", location, message);
+  } else {
+    FPrintF(stderr, "FATAL ERROR: %s\n", message);
+  }
+
+  Isolate* isolate = Isolate::TryGetCurrent();
+  Environment* env = nullptr;
+  if (isolate != nullptr) {
+    env = Environment::GetCurrent(isolate);
+  }
+  bool report_on_fatalerror;
+  {
+    Mutex::ScopedLock lock(node::per_process::cli_options_mutex);
+    report_on_fatalerror = per_process::cli_options->report_on_fatalerror;
+  }
+
+  if (report_on_fatalerror) {
+    report::TriggerNodeReport(
+        isolate, env, message, "OOMError", "", Local<Object>());
```

Comment:
I think we should either use `MaybeLocal<Object>()` here, or continue using `Local<Object>` in the functions (that defaults to `Local<Object>()` or with a comment describing what empty handles mean) and check if the handle is empty instead. The latter is probably more consistent with V8's API conventions though.

Reply:
Do you suggest sticking with `Local<Object>` in function params?

Reply:
That would be my preference, yes

Reply:
Thank you for the idea. Sounds good to me. I've reverted this type of change.